codecov.yaml

```diff
@@ -1,53 +1,70 @@
 codecov:
+  # The default branch is used to identify:
+  # Which branch to cache the repository YAML for UI changes.
+  # Which branch is the first branch on the repository dashboard in Codecov.
   branch: main
+  # Show the Codecov status without waiting for other status checks to complete or pass.
   require_ci_to_pass: no
 
 comment:
-  layout: "reach,diff,flags,tree"
+  # Customize comment layout by choosing the order and types of components to include.
+  # `Reach` is a coverage graph embedded into the comment.
+  # `Diff` is the Coverage Diff of the pull request.
+  # `Flags` are a list of user defined Flags, and the impact on their coverage. refs: https://docs.codecov.com/docs/flags
+  # `Files` or Tree are a list of files that are impacted by the pull request (coverage changes, file is new or removed).
+  # `Header` is the top of the comment showing the two commits being merged, the change in coverage, and the diff coverage.
+  # `Footer` is the bottom of a comment that contains the legend, last updated commits, and docs link.
+  layout: "reach,diff,flags,tree,header,footer"
+  # The way Codecov submits comments in your pull requests.`default` means update if exists.Otherwise, post new.
   behavior: default
-  require_changes: no
+  # Comments will now only post when coverage changes. Furthermore, if a comment already exists, and a newer commit results in no coverage change for the entire pull, the comment will be deleted.
+  require_changes: true
+  # Force comments to post on pull requests even if Codecov doesn't have coverage reports for either the base or head commit.
   require_base: no
   require_head: no
-  after_n_builds: 1
-  show_carryforward_flags: true
+  # Codecov do not comment until at least 5 builds have been uploaded
+  # from the CI pipeline. By default, codecov will post and/or update the
+  # pull request comment after it processes each report uploaded for a
+  # particular pull request commit. As the Antrea CI process uploads many
+  # reports to codecov, this can be confusing for team members viewing the
+  # pull request as reports are processing.
+  after_n_builds: 5
```

Comment:
how was 5 chosen?

Reply:
In the go.yaml
1. test-unit-ubuntu
2. test-unit-windows
3. test-integration
In the kind.yaml
4. test-e2e-encap
5. test-e2e-encap-no-proxy
6. test-e2e-encap-all-features-enabled
7. test-e2e-noencap
8. test-e2e-hybrid
9. test-e2e-flow-visibility
It seems there are 8 uploads for each commit, as I see on the Codecov report page.

Reply:
I am wondering if there is only 3 builds( reports) for the PR, will it update the report comment after all 3 builds finished? or it has to be at least 5 builds?

Reply:
Iit has to be at least 5 builds.